java/opendcs/src/main/java/decodes/tsdb/algo/WeightedAverage.java

```diff
@@ -0,0 +1,110 @@
+package decodes.tsdb.algo;
+
+import decodes.tsdb.DbCompException;
+import ilex.var.NamedVariable;
+import org.opendcs.annotations.algorithm.Algorithm;
+import org.opendcs.annotations.algorithm.Input;
+import org.opendcs.annotations.algorithm.Output;
+
+@Algorithm(description = "Calculate Weighted Average values (eg Water Temperature) using: sum(input1n*input2n) / total2")
+public class WeightedAverage
+	extends AW_AlgorithmBase
+{
+	@Input
+	public double weightTotal;
```

Comment:
it seems like it would be better to calculate this instead of have it, but that may depend on a question below as to whether the weights are time series or not.

Reply:
This is definitely a timeseries that the user supplies. It does behave poorly if the total doesn't match the sum of the weights though!